### PR TITLE
possible fix for mac error

### LINKED ISSUE #199: [BUG] MacOS pytest doesn't work out of the box
**Describe the bug**
`pytest` doesn't work on `pip install -e ".[dev]" in clean conda env.
I get
```bash
...
exec(co, module.__dict__)
tests/tests/conftest.py:8: in <module>
    from cellfinder_core.download import models
src/cellfinder_core/__init__.py:13: in <module>
    raise PackageNotFoundError(
E   importlib.metadata.PackageNotFoundError: No package metadata was found for cellfinder tools cannot be invoked without tensorflow. Please install tensorflow into your environment to use cellfinder tools. For more information, please see https://github.com/brainglobe/brainglobe-meta#readme.
```

My guess is this is because our guard checks for the existence of a `tensorflow` package, but TensorFlow is called `tensorflow-macos` on Macs. 

I could be missing something though. I'm surprised we haven't noticed this elsewhere 🤔 

**To Reproduce**
See above: `pip install -e ".[dev]"` in clean conda env on Apple Silicon.

**Expected behaviour**
`pytest` should run the tests out-of-the box in a dev environment 

**Log file**
n/a

**Screenshots**
n/a

**Computer used (please complete the following information):**

- MacOS Apple M2

### PR DESCRIPTION
**Pull request by alessandrofelder**
Before submitting a pull request (PR), please read the [contributing guide](https://github.com/brainglobe/.github/blob/main/CONTRIBUTING.md).

## Description

**What is this PR**

- [x] Bug fix
- [ ] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
I had some troubles installing a clean development environment on my macbook.

**What does this PR do?**
extends the `tensorflow` guard to include `tensorflow-macos`.

## References

Closes #199 

## How has this PR been tested?
Works locally for me?

## Is this a breaking change?
Nope

## Does this PR require an update to the documentation?
Nope

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality (unit & integration)
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Review by willGraham01 — COMMENTED** [review]
Test failure seems genuine but I don't believe that it would be caused by any changes in this PR (the tests run on `ubuntu-latest`...).

Going forward, your thoughts on the value of us having a separate workflow that checks the `tensorflow` catch is behaving itself? Something like the following in a conda environment:

```sh
$ pip install .
$ python import cellfinder-core
# should be OK
$ pip uninstall tensorflow
$ python import cellfinder-core
# should error
```

File: src/cellfinder_core/__init__.py [hunk 1]
```diff
@@ -10,12 +10,15 @@
...
     TF_VERSION = version("tensorflow")
 except PackageNotFoundError as e:
-    raise PackageNotFoundError(
-        f"cellfinder tools cannot be invoked without tensorflow. "
-        f"Please install tensorflow into your environment to use cellfinder tools. "
-        f"For more information, please see "
-        f"https://github.com/brainglobe/brainglobe-meta#readme."
-    ) from e
+    try:
+        TF_VERSION = version("tensorflow-macos")
```
[the remark this answers is not in the document]
Comment (alessandrofelder) [reply src/cellfinder_core/__init__.py:14]: Hopefully we won't need this in the future (when we move to later `tensorflow` versions that can be installed on MacOS the same way as everywhere else, or move away from `tensorflow\ entirely) but good to have for now!

**Comment by alessandrofelder** [comment]
I think having that extra workflow as a test is a great idea!

**Comment by alessandrofelder** [comment]
> Test failure seems genuine but I don't believe that it would be caused by any changes in this PR (the tests run on `ubuntu-latest`...).
Yea, the test fails in the same way in a completely different PR too. And it's the cellfinder, not the cellfinder-core test that fails. Should be addressed in a separate PR?

**Review by willGraham01 — APPROVED** [review]
> I think having that extra workflow as a test is a great idea!

Have broken this out into #203. I can take a look at this myself next week when I'm back on BG full-time.

> Yea, the test fails in the same way in ahttps://github.com/brainglobe/cellfinder-core/pull/198 And it's the cellfinder, not the cellfinder-core test that fails. Should be addressed in a separate PR? I've opened https://github.com/brainglobe/cellfinder-core/issues/202 for this

Cool, providing approval now :+1:

**Pull request merged**